functions.php

```diff
@@ -1319,3 +1340,271 @@ function wp_autoupdates_themes_bulk_actions_handle( $redirect_to, $doaction, $it
+						array(
+							'enabled_count'  => '(' . $enabled_count . ')',
+							'disabled_count' => '(' . absint( count( $all_themes ) - $enabled_count ) . ')',
+							'return_html'    => $return_html,
+							'type'           => 'theme',
+						)
+					);
+					break;
+				}
+			}
+		}
+	}
+	wp_send_json_error(
+		array(
+			'error' => __( 'Could not disable auto-updates for the selected item.', 'wp-autoupdates' ),
+		)
+	);
+}
+add_action( 'wp_ajax_disable_auto_updates', 'wp_autoupdates_disable_auto_updates' );
+/**
+ * Enable auto updates via Ajax.
+ */
+function wp_autoupdates_enable_auto_updates() {
+	$type  = sanitize_text_field( $_POST['type'] );
+	$asset = sanitize_text_field( urldecode( $_POST['asset'] ) );
+	check_ajax_referer(
+		sprintf(
+			'autoupdate-%s_%s',
+			$type,
+			$asset
+		)
+	);
+
+	// Capability check.
+	if ( 'plugin' === $type ) {
+		if ( ! current_user_can( 'update_plugins' ) ) {
+			wp_send_json_error(
+				array(
+					'error' => __( 'You do not have permission to modify plugins.', 'wp-autoupdate' ),
+				)
+			);
+		}
+	}
+	if ( 'theme' === $type ) {
+		if ( ! current_user_can( 'update_themes' ) ) {
+			wp_send_json_error(
+				array(
+					'error' => __( 'You do not have permission to modify themes.', 'wp-autoupdate' ),
+				)
+			);
+		}
+	}
+
+	// Check Asset Type.
+	if ( 'plugin' === $type ) {
+		$wp_autoupdate_plugins = get_site_option( 'wp_auto_update_plugins', array() );
+		if ( is_array( $wp_autoupdate_plugins ) ) {
+			$wp_autoupdate_plugins[] = $asset;
+			array_unique( $wp_autoupdate_plugins );
+			update_site_option( 'wp_auto_update_plugins', $wp_autoupdate_plugins );
+
+			/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
+			$all_plugins           = apply_filters( 'all_plugins', get_plugins() );
+			$wp_autoupdate_plugins = array_intersect( $wp_autoupdate_plugins, array_keys( $all_plugins ) );
+			$enabled_count         = count( $wp_autoupdate_plugins );
+			$plugin_name           = $all_plugins[ $asset ]['Name'];
+
+			$plugin_url = add_query_arg(
+				array(
+					'action'   => 'autoupdate',
+					'plugin'   => $asset,
+					'_wpnonce' => wp_create_nonce( 'autoupdate-plugin_' . $asset ),
+				),
+				'plugins.php'
+			);
+
+			$aria_label = esc_attr(
+				sprintf(
+					/* translators: Plugin name. */
+					_x( 'Enable automatic updates for %s', 'plugin', 'wp-autoupdates' ),
+					esc_html( $plugin_name )
+				)
+			);
+			$return_html  = '<p>';
+			$return_html .= '<span class="plugin-autoupdate-enabled">' . esc_html__( 'Auto-updates enabled', 'wp-autoupdates' ) . '</span>';
+			$return_html .= '<br />';
+			$return_html .= sprintf(
+				'<a href="%s" class="plugin-autoupdate-disable" aria-label="%s">%s</a>',
+				esc_url_raw( $plugin_url ),
+				$aria_label,
+				esc_html__( 'Disable auto-updates', 'wp-autoupdates' )
+			);
+			$return_html .= '</p>';
+
+			wp_send_json_success(
+				array(
+					'enabled_count'  => '(' . $enabled_count . ')',
```

Comment:
Let's add an `absint` here too.

```suggestion
					'enabled_count'  => '(' . absint( $enabled_count ) . ')',
```